modules/ROOT/pages/_partials/sync/server-getting-started-with-an-example-data-source.adoc

```diff
@@ -0,0 +1,133 @@
+[id="server-getting-started-with-an-example-data-source-{context}"]
+= Getting started with an example data source
+
+This section describes how to create a GraphQL schema for an existing data source schema.
+
+For this procedure, the following describes the existing data source schema:
+
+[.Employee]
+|===
+| employee_id, `ID`
+| employee_name, `String`
+| category_id, `ID`
+|===
+
+[.Category]
+|===
+| category_id, `ID`
+| employee_type, `String`
+| security_clearance, `ID`
+|===
+
+.Prerequisites
+
+* xref:getting-started-with-hello-world-{context}[].
+* You are familiar with {sync-server} operations, for example, stopping and starting {sync-server}.
+
+.Procedure
+
+. If the {sync-server} is running, stop it.
++
+. Edit the `index.js` file and replace its contents with the following content:
++
+[source,javascript]
+----
+
+const express = require('express')
+//Include our server libraries
+const { VoyagerServer, gql } = require('@aerogear/voyager-server')
+
+// Stub that reflects data source mapping
+const employees = [{employee_id: 1, employee_name: "joe", cateory_id: 1 }, {employee_id: 2, employee_name: "john", cateory_id: 2}];
+
+const categories = [{category_id: 1, employee_type:"Manager", security_clearance: 1}, {category_id: 2, employee_type:"Associate", security_clearance: 2}];
+
+//GraphQL schema for mapping to data source mapping
+const typeDefs = gql`
+  type Employee {
+      employee_id: ID!,
+      employee_name: String!
+      category: Category
+  }
+  type Category {
+      category_id: ID!,
+      employee_type: String!,
+      security_clearance: ID!
+  }
+  type Query {
+    listEmployees: [Employee]
+  }
+`
+
+//Create the resolvers for your schema
+const resolvers = {
+  Query: {
+    listEmployees: (obj, args, context, info) => {
+      return employees;
+    }
+  },
+  Employee: {
+    category(employee) {
+        return filter(categories, {category_id: employee.cateory_id});
```

Comment:
Where is the `filter` function defined/imported? I guess [it comes from `lodash` module](https://www.apollographql.com/docs/apollo-server/essentials/data#resolver-map), but it's missing in this code snippet, so it could be a bit confusing...

Reply:
TBH, I'm not sure :)

Do you think we can use it to display more data as part of this?

. To list Employees, execute the following command in the GraphQL window:
+
[source,javascript]
----
{
  listEmployees
  {
		employee_id,
    employee_name,
  }
}
----

I'd love to update ^ to display category info too (irrespective of whether the "filter" code is applicable).